### PR TITLE
Bhbc 743

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- {https://quartech.atlassian.net/browse/BHBC-743}

## This PR contains the following changes

- {UI changes including BC gov styles, project list to match the list in the DB, unit test to /app, unit test and light integration test for /api}

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Bug fix (change which fixes an issue)
- [x] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [x] New and existing unit tests pass locally with my changes

### Code

- [x] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [x] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [x] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [x] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [x] My code changes account for null/undefined values and handle errors appropriately
- [x] My code uses types/interfaces to help describe values/parameters/etc and inform future developers

### Documentation
- [x] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation

### Tests
- [x] I have added/updated unit tests for any new code I've added/updated
- [ ] I have added/updated Postman tests to account for any new/updated code

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- built unit test
- started on an integration test, but commented it out, to be migrated to app level integration testing.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/paths/projects.inttest.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+export {};
```
Comment (NickPhura) [inline api/src/paths/projects.inttest.ts]: The filename will need to be `*.test.ts` or it won't get picked up by the test runner

File: api/src/paths/projects.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { expect } from 'chai';
```
Comment (NickPhura) [inline api/src/paths/projects.unit.test.ts]: Can we move these tests into the other test file? We only need to maintain 1 test file (per code file), that can hold all of the tests.

File: api/src/paths/projects.inttest.ts (hunk 1 above)
Comment (anissa-agahchen) [reply api/src/paths/projects.inttest.ts]: I changed that intentionally in other for the integration test to not run ... as I want to rule out the integration test as the issue ... let's chat.

File: api/src/paths/projects.unit.test.ts (hunk 2 above)
Comment (anissa-agahchen) [reply api/src/paths/projects.unit.test.ts]: yes

File: app/src/themes/appTheme.ts [hunk 3]
```diff
@@ -43,6 +47,14 @@ const appTheme = createMuiTheme({
...
       }
     },
+    // typography: {
+    //   body: {
+    //     fontFamily: "'Open Sans', sans-serif",
+    //     fontWeight: 400,
+    //     fontSize: 16,
+    //     color: "red"
+    //   }
+    // },
```
Comment (NickPhura) [inline app/src/themes/appTheme.ts]: Looks like this is captured above, may as well remove this.

File: app/src/themes/appTheme.ts [hunk 4]
```diff
@@ -21,6 +22,9 @@ const appTheme = createMuiTheme({
       disabled: 'rgba(0, 0, 0, 0.67)'
     }
   },
+  typography: {
+    fontFamily: ['BCSans'].join(',')
```
Comment (NickPhura) [inline app/src/themes/appTheme.ts:26]: Nice!

File: env_config/env.docker [hunk 5]
```diff
@@ -37,8 +37,8 @@ DB_SCHEMA_DAPI_V1=biohub_dapi_v1
 # ------------------------------------------------------------------------------
 # KeyClock URLS
 # ------------------------------------------------------------------------------
+# KEYCLOAK_URL=https://dev.oidc.gov.bc.ca/auth/realms/dfmlcg7z/protocol/openid-connect/certs
```
Comment (NickPhura) [inline env_config/env.docker]: Extra line, can remove

File: api/src/utils/template-utils.test.ts [hunk 6]
```diff
@@ -29,7 +29,15 @@ describe('isValidJSONSchema', () => {
   });
 
   describe('with a valid schema', () => {
-    const jsonSchema = {};
+    const jsonSchema = {
+      title: 'An Invalid JSONSchema',
```
Comment (NickPhura) [inline api/src/utils/template-utils.test.ts]: nit: can you change the title to 'A Valid JSONSchema' ?

File: env_config/env.docker (hunk 5 above)
Comment (anissa-agahchen) [reply env_config/env.docker]: done

File: app/src/themes/appTheme.ts (hunk 3 above)
Comment (anissa-agahchen) [reply app/src/themes/appTheme.ts]: done

File: app/src/AppRouter.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import HomeRouter from 'features/home/HomeRouter';
+//import ProjectsRouter from 'features/home/HomeRouter';
```
Comment (NickPhura) [inline app/src/AppRouter.tsx]: nit: delete line

Can delete the entire `/home` folder as well as its not being used, and doesn't have anything worth keeping in it.

File: app/src/components/tabs/TabsComponent.tsx [hunk 8]
```diff
@@ -35,7 +35,7 @@ const TabsComponent: React.FC = () => {
 
   return (
     <Tabs value={activeTab} onChange={handleChange}>
-      <Tab label="Home" onClick={() => history.push('/home')} />
+      {/* <Tab label="Home" onClick={() => history.push('/home')} /> */}
```
Comment (NickPhura) [inline app/src/components/tabs/TabsComponent.tsx]: nit: delete

File: app/src/themes/appTheme.ts (hunk 3 above)
Comment (NickPhura) [reply app/src/themes/appTheme.ts]: Looks like this got missed

File: env_config/env.docker (hunk 5 above)
Comment (NickPhura) [reply env_config/env.docker]: Looks like this got missed

File: api/src/queries/project-queries.ts [hunk 9]
```diff
@@ -127,10 +123,11 @@ export const getProjectsSQL = (projectId: string): SQLStatement | null => {
...
       scientific_collection_permit_number,
       management_recovery_action,
-      start_date,
-      end_date
+      to_char(start_date,'MM/DD/YYYY') as start_date,
+      to_char(end_date,'MM/DD/YYYY') as end_date,
+      location_description
     from
-      project;
+      biohub.project;
```
Comment (NickPhura) [inline api/src/queries/project-queries.ts]: With the context fix PR I put up we shouldn't need to specify schema anymore, the biohub_api user will have the schemas already set on its search path

**Review by rstens — APPROVED** [review]

**Pull request merged**